### PR TITLE
Optionally specify limit for number of entities in a record.

### PR DESCRIPTION
**Pull request by blackwinter**
_Just a proof-of-concept to gauge interest in such a feature; the immediate need has been rendered moot (due to the record's holdings being suppressed from publishing now)._

This is a brute-force approach to dealing with OOM situations when Alma records have an excessive number of items (e.g. [99374518570506441](https://lobid.org/resources/99374518570506441): >12000 entities = ~10 GB heap for the Record instance). Loosely modeled after JAXP's [`totalEntitySizeLimit`](https://docs.oracle.com/en/java/javase/13/security/java-api-xml-processing-jaxp-security-guide.html#GUID-82F8C206-F2DF-4204-9544-F96155B1D258__TABLE_RQ1_3PY_HHB), except that we don't abort processing but instead skip the remaining entities.
    
~Use Metafix instance setter `setMaxEntityCount(int)` or~ set system property `org.metafacture.metafix.maxEntityCount=<int>`.
    
Alternative options:
    
- Increase maximum heap size for JVM.
- Significantly reduce memory requirement for Record instances.

Possible optimizations (implemented in dc4f1af):

1. Drop the setter and make the limit `static final`; this way the JVM can (potentially) eliminate the check altogether when no limit has been set (less than zero).
2. Skip _all_ remaining literals (instead of just the ones inside entities), thereby simplifying the check in `literal()`.

### CODE REVIEW

**Review by dr0i — APPROVED** [review]
I think this looks good. It prevents possible flooding, as we already experienced. So I like this feature.

File: metafix/src/main/java/org/metafacture/metafix/Metafix.java [hunk 1]
```diff
@@ -313,22 +315,36 @@ public void startEntity(final String name) {
             throw new IllegalArgumentException("Entity name must not be null.");
         }
 
+        ++entityCount;
+        if (maxEntityCountExceeded()) {
+            LOG.debug("Maximum number of entities exceeded: {}/{}", entityCount, MAX_ENTITY_COUNT);
```
[the remark this answers is not in the document]
Comment (fsteeg) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java:320]: We could add a field, and only log when it's set for the first time, but since this is on `debug` level I think it's fine. Also, it might be useful to see how many entities were actually skipped.

**Review by fsteeg — COMMENTED** [review]
Looks good, but should probably include a test.

**Comment by blackwinter** [comment]
Regarding test: As currently implemented, the feature cannot be exercised under test conditions because the limit is already set (system property already evaluated) before the test gets the chance to influence it. Either change the implementation to make it testable (keeping performance implications in mind) or skip test coverage in this case (assuming that this feature is only used in extremely rare edge cases).

**Comment by blackwinter** [comment]
Adding a unit test proved to be quite messy when attempting to keep the instance member variable `final`. Added an integration test instead (a10b3d8). This has the added benefit that it actually exercises the retrieval of the system property.
